tests/apollo/test_skvbc_chaotic_startup.py

```diff
@@ -0,0 +1,271 @@
+                        break
+
+        except Exception as e:
+            print(f"Delayed replicas start-up failed for start-up order: {replicas_starting_order}")
+            raise e
+        else:
+            print(f"Delayed replicas start-up succeeded for start-up order: {replicas_starting_order}. "
+                  f"The BFT network eventually stabilized in view #{current_view}.")
+
+    @with_trio
+    @with_bft_network(start_replica_cmd_with_vc_timeout("20000"),
+                      selected_configs=lambda n, f, c: f >= 2)
+    @with_constant_load
+    async def test_f_staggered_replicas_requesting_vc(self, bft_network, skvbc, nursery):
+        """
+        The goal of this test is to verify correct behaviour in the situation where
+        a subset of F replicas not big enough to reach quorum for execution starts early,
+        accepts client requests and initiates view change. The correct behaviour in this
+        case will be to for the system to make progress in the slow path after the remaining
+        2F+1 replicas are up. After execution is confirmed in the slow path, we stop one
+        replica, but not the current or next Primary to verify View Change will happen due
+        to inability to reach quorums for execution in the current View. When the new View
+        is active we verify the system is able to make progress.
+
+        1) We use a decorator providing constant load of client requests from the start of the test.
+        2) Given a BFT network, we start F out of all N replicas.
+        3) We wait for them to initiate View Change.
+        4) Then we start the remaining 2F+1 replicas.
+        5) We verify the system is making progress until a Checkpoint is reached.
+        6) Then we verify the early F replicas that initiated View Change catchup by State Transfer.
+        7) After this we stop 1 replica, but not the initial or expected next primary.
+        8) We verify View Change happens and the system is able to execute requests.
+        """
+
+        n = bft_network.config.n
+        f = bft_network.config.f
+        c = bft_network.config.c
+
+        initial_view = initial_primary = 0
+        expected_next_view = expected_next_primary = 1
+
+        # take a random set containing F replicas out of all N without the initial primary
+        early_replicas = set()
+        for _ in range(f):
+            exclude_replicas = early_replicas | {initial_primary}
+            early_replicas.add(random.choice(bft_network.all_replicas(without=exclude_replicas)))
+
+        late_replicas = bft_network.all_replicas(without=early_replicas)
+
+        print(f"STATUS: Starting F={f} replicas.")
+        bft_network.start_replicas(replicas=early_replicas)
+
+        print("STATUS: Wait for early replicas to initiate View Change.")
+        for r in early_replicas:
+            active_view_of_replica = 0
+            view_of_replica = 0
+            while active_view_of_replica == view_of_replica:
```

Comment:
If the metric client is timedout, then we may end up with active_view_of_replica=None and view_of_replica != None